modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -215,26 +216,31 @@ private void initConfigurationResolver() {
 
     Optional<ConfigurationInstance> staticConfiguration = getStaticConfiguration();
     if (staticConfiguration.isPresent()) {
-      configurationResolver = event -> staticConfiguration;
+      configurationResolver = (event, stopping) -> staticConfiguration;
       return;
     }
 
     if (isConfigurationSpecified()) {
       // the config is dynamic
-      configurationResolver = event -> {
-        ConfigurationInstance instance = configurationProvider.get().get(event);
-        if (instance == null) {
-          throw new IllegalModelDefinitionException(format(
-                                                           "Root component '%s' contains a reference to config '%s' but it doesn't exists",
-                                                           getLocation().getRootContainerName(),
-                                                           configurationProvider));
-        }
+      configurationResolver = (event, stopping) -> {
+        if (stopping) {
+          // Avoid populating the cache of dynamic connections if stopping
+          return configurationProvider.get().getIfPresent(event);
```

Comment:
This isn't a matter of not populating the cache. If the context is stopping, the dynamic config shouldn't be spawned at all.